lib/scholar/linear/svm.ex

```diff
@@ -0,0 +1,250 @@
+      end
+
+    n = Nx.axis_size(x, -1)
+    num_classes = opts[:num_classes]
+
+    coef =
+      Nx.broadcast(
+        Nx.tensor(1.0, type: to_float_type(x)),
+        {num_classes, n}
+      )
+
+    bias = Nx.broadcast(Nx.tensor(0, type: to_float_type(x)), {num_classes})
+
+    coef_optimizer_state = optimizer_init_fn.(coef) |> as_type(to_float_type(x))
+    bias_optimizer_state = optimizer_init_fn.(bias) |> as_type(to_float_type(x))
+
+    opts = Keyword.put(opts, :optimizer_update_fn, optimizer_update_fn)
+
+    fit_n(x, y, coef, bias, coef_optimizer_state, bias_optimizer_state, opts)
+  end
+
+  deftransformp as_type(container, target_type) do
+    Nx.Defn.Composite.traverse(container, fn t ->
+      type = Nx.type(t)
+
+      if Nx.Type.float?(type) and not Nx.Type.complex?(type) do
+        Nx.as_type(t, target_type)
+      else
+        t
+      end
+    end)
+  end
+
+  # SVM training loop
+
+  defnp fit_n(x, y, coef, bias, coef_optimizer_state, bias_optimizer_state, opts) do
+    iterations = opts[:iterations]
+    num_classes = opts[:num_classes]
+    optimizer_update_fn = opts[:optimizer_update_fn]
+    c = opts[:c]
+    eps = opts[:eps]
+    margin = opts[:margin]
+
+    {{final_coef, final_bias}, _} =
+      while {{coef, bias},
+             {x, iterations, y, coef_optimizer_state, bias_optimizer_state,
+              has_converged = Nx.broadcast(Nx.u8(0), {num_classes}), eps, margin, c, iter = 0}},
+            iter < iterations and not Nx.all(has_converged) do
+        # ++++ inner while ++++++
+        {{coef, bias, has_converged, coef_optimizer_state, bias_optimizer_state}, _} =
+          while {{coef, bias, has_converged, coef_optimizer_state, bias_optimizer_state},
+                 {x, y, c, margin, iterations, iter, eps, j = 0}},
+                j < num_classes do
+            y_j = y == j
+            coef_j = Nx.take(coef, j)
+            bias_j = Nx.take(bias, j)
+
+            {loss, {coef_grad, bias_grad}} = loss_and_grad(coef_j, bias_j, x, y_j, c, margin)
+            grad = Nx.broadcast(0.0, {num_classes, Nx.axis_size(x, 1)})
+            coef_grad = Nx.put_slice(grad, [j, 0], Nx.new_axis(coef_grad, 0))
+
+            {coef_updates, coef_optimizer_state} =
+              optimizer_update_fn.(coef_grad, coef_optimizer_state, coef)
+
+            coef = Polaris.Updates.apply_updates(coef, coef_updates)
+
+            grad = Nx.broadcast(0.0, {num_classes})
+            bias_grad = Nx.put_slice(grad, [j], Nx.new_axis(bias_grad, 0))
+
+            {bias_updates, bias_optimizer_state} =
+              optimizer_update_fn.(bias_grad, bias_optimizer_state, bias)
+
+            bias = Polaris.Updates.apply_updates(bias, bias_updates)
+
+            has_converged_j = Nx.sum(Nx.abs(loss)) < Nx.axis_size(x, 0) * eps
+
+            has_converged =
+              Nx.indexed_put(
+                has_converged,
+                Nx.new_axis(Nx.new_axis(j, -1), -1),
+                Nx.new_axis(has_converged_j, -1)
```

Comment:
I think we can use this now:
```suggestion
                Nx.new_axis(j, -1),
                has_converged_j
```